lib/internal/Magento/Framework/Reflection/TypeProcessor.php

```diff
@@ -708,4 +695,58 @@ private function getNormalizedType($type)
         }
         return $type;
     }
+
+    /**
+     * Parses `return` annotation from reflection method.
+     *
+     * @param MethodReflection $methodReflection
+     * @return ReturnTag
+     * @throws \InvalidArgumentException if doc block is empty or `@return` annotation doesn't exist
+     */
+    private function getMethodReturnAnnotation(MethodReflection $methodReflection)
+    {
+        $methodName = $methodReflection->getName();
+        $returnAnnotations = $this->getReturnFromDocBlock($methodReflection);
+        if (empty($returnAnnotations)) {
+            // method can inherit doc block from implemented interface, like for interceptors
+            $implemented = $methodReflection->getDeclaringClass()->getInterfaces();
+            /** @var ClassReflection $parentClassReflection */
+            foreach ($implemented as $parentClassReflection) {
+                if (!$parentClassReflection->hasMethod($methodName)) {
+                    continue;
+                }
+                $returnAnnotations = $this->getReturnFromDocBlock(
+                    $parentClassReflection->getMethod($methodName)
+                );
+
+                break;
```

Comment:
Currently logic of this cycle is not so easy to understand. I suggest something like

```
foreach (...) {
    if ($parent...->hasMethod()) {
        $returnAnnotations = ...;
        break;
    }
}
```

Reply:
Done

Reply:
Don't see it :(

Reply:
Please, see a list of changed files, I amended commit.

Reply:
Hm, I was answering to next thread actually O.o